components/sup/src/manager/service/mod.rs

```diff
@@ -926,10 +986,46 @@ impl Service {
                 (false, _) => HealthCheck::Critical,
             }
         };
+        if check_result == HealthCheck::Ok {
+            self.schedule_routine_health_check();
+            debug!(
+                "Service ({}) health check is: {}",
+                self.spec_ident, check_result
+            );
+        } else {
+            self.schedule_special_health_check();
+            debug!(
+                "Service ({}) health check is: {}. Re-running within default {:#?}",
+                self.spec_ident,
+                check_result,
+                *HealthCheckInterval::default().as_ref()
```

Comment:
```suggestion
                 HealthCheckInterval::default()
```

Reply:
Yes, good catch! Now that I implemented `fmt::Display for HealthCheckInterval` that will work and is easier on the eyes.